workflow/controller/operator.go

```diff
@@ -2026,6 +2026,22 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 			// Last child node is still running.
 			nodeName = lastChildNode.Name
 			node = lastChildNode
+
+			// Instance may be missing caused by creation failure, for example, ResourceQuota error,
+			// so when recreating the pod, we also need to inject variables.
+			woc.log.Debugf("Assume instance is missing, nodeName: %s, all children: %v", nodeName, retryParentNode.Children)
+			localParams := make(map[string]string)
+			// Change the `pod.name` variable to the retry node name
+			if processedTmpl.IsPodType() {
+				localParams[common.LocalVarPodName] = woc.getPodName(nodeName, processedTmpl.Name)
+			}
+			// Inject the retryAttempt number
+			localParams[common.LocalVarRetries] = strconv.Itoa(len(retryParentNode.Children) - 1)
```

Comment:
I don't know this code super well, but why is this one subtracting 1?

Reply:
If lastChildNode is nil, it will be initialized and added to node status. is it is not nil, indicating we have added it in the previous attempt, because retries variable equals node index minus 1, so minus 1 here.

Reply:
I see, thanks. I'd misinterpreted `LocalVarRetries` to mean number of retries instead of retry index.